app/controllers/publishers/vacancies/documents_controller.rb

```diff
@@ -29,12 +29,12 @@ def destroy
   end
 
   def confirm
-    if confirmation_form.valid?
-      return redirect_to_next_step unless uploading_more_documents?
+    return render :index unless confirmation_form.valid?
 
+    if uploading_more_documents?
       redirect_to new_organisation_job_document_path(vacancy.id)
     else
-      render :index
+      redirect_to_job_summary
```

Comment:
These changes fix the `redirect_to_next_step` wrongly sending the user to the "upload an additional document" page even when they answer "NO" to the "do you want to upload an additional document?" question.

We cannot rely on its logic for this case, as the documents form without a document is no longer valid, so we do an explicit redirection to the right step instead.